nixos/modules/services/security/authentik.nix

```diff
@@ -0,0 +1,233 @@
+      default = 9443;
+      description = "The https port that authentik will listen on.";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "authentik";
+      description = "The user immich should run as.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "authentik";
+      description = "The group immich should run as.";
+    };
+    database = {
+      enable =
+        mkEnableOption "the postgresql database for use with authentik. See {option}`services.postgresql`"
+        // {
+          default = true;
+        };
+      createDB = mkEnableOption "the automatic creation of the database for authentik." // {
+        default = true;
+      };
+      name = mkOption {
+        type = types.str;
+        default = "authentik";
+        description = "The name of the authentik database.";
+      };
+      host = mkOption {
+        type = types.str;
+        default = "/run/postgresql";
+        example = "127.0.0.1";
+        description = "Hostname or address of the postgresql server. If an absolute path is given here, it will be interpreted as a unix socket path.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = "The port of the postgresql server";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "authentik";
+        description = "The database user for authentik.";
+      };
+    };
+    redis = {
+      enable = mkEnableOption "a redis cache for use with authentik" // {
+        default = true;
+      };
+      host = mkOption {
+        type = types.str;
+        default = config.services.redis.servers.authentik.unixSocket;
+        defaultText = lib.literalExpression "config.services.redis.servers.authentik.unixSocket";
+        description = "The host that redis will listen on.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 0;
+        description = "The port that redis will listen on. Set to zero to disable TCP.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.database.createDB -> cfg.database.name == cfg.database.user;
+        message = "The postgres module requires the database name and the database user name to be the same.";
+      }
+      {
+        assertion = cfg.secretsFile == null -> lib.hasPrefix "file://" cfg.environment.AUTHENTIK_SECRET_KEY;
+        message = "If no secretsFile is provided, services.authentik.environment has to provide AUTHENTIK_SECRET_KEY with a file:// prefix";
+      }
+      {
+        assertion = cfg.secretsFile != null -> !(cfg.environment ? "AUTHENTIK_SECRET_KEY");
+        message = "Possibly conflicting AUTHENTIK_SECRET_KEY definition in secretsFile and services.authentik.environment";
+      }
+    ];
+
+    systemd.services.authentik-server = {
+      description = "Authentik server component";
+      after = [ "network.target" ];
```

Comment:
This only indicates that the networking stack has been _started_. For almost all services - this one included - it makes more sense to use `network-online.target`.